pkg/repository/git.go

```diff
@@ -113,12 +129,55 @@ type oktetoGitWorktree struct {
 	worktree *git.Worktree
 }
 
-func (ogr oktetoGitWorktree) Status() (gitStatusInterface, error) {
-	status, err := ogr.worktree.Status()
+func (ogr oktetoGitWorktree) GetRoot() string {
+	return ogr.worktree.Filesystem.Root()
+}
+
+func fixDubiousOwnershipConfig(path string) error {
+	c := exec.Command("git", "config", "--global", "--add", "safe.directory", path)
+	c.Dir = path
+	return c.Run()
+}
+
+func runGitStatusCommand(path string, fixAttempt int) (string, error) {
+	if fixAttempt > 0 {
+		return "", fmt.Errorf("failed to get status: too many attempts")
+	}
+	c := exec.Command("git", "status", "--porcelain", "-z")
+	c.Dir = path
+	output, err := c.Output()
+
+	if errors.Is(err, errors.New("detected dubious ownership in repository")) {
+		err = fixDubiousOwnershipConfig(path)
+		if err != nil {
+			return "", fmt.Errorf("failed to get status: cannot recover")
+		}
+		fixAttempt++
+		return runGitStatusCommand(path, fixAttempt)
+	}
+
+	return string(output), err
+}
+
+func (ogr oktetoGitWorktree) Status() (oktetoGitStatus, error) {
+	output, err := runGitStatusCommand(ogr.GetRoot(), 0)
```

Comment:
the first iteration removed the old logic, but I am considering keeping the git status from `git-go` so that we can keep the CLI backward compatible (ie. do not require the user/host to have the `git` binary installed).

The logic would be:

- try find `git`, if present, then use the new fast approach
- if `git` is missing, then fallback to `git-go`

Reply:
Let's do that and go with the fallback. Also add a log (`oktetoLog.Infof()`) informing that we are using the old method then